### PR TITLE
Put request

### PR DESCRIPTION
**Pull request by grace000**
This PR covers [this story](https://trello.com/c/CuwJJN2b/29-put-request-375).

Client can now make changes to a file on the server with plain text and make a GET request to see those changes.

### CODE REVIEW

File: src/main/java/github/grace000/erversay/RouteHandlers/Put.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private ResponseBuilder responseBuilder = new ResponseBuilder();
+    private String absolutePath = "public/sample.txt";
+    private List<String> resource = new ArrayList<>();
+    private Response response;
+
+    public enum AcceptedMethods {
+       GET, PUT
+    }
+
+    public boolean isMethodAllowed(String method) {
```
Comment (ksabbak) [inline src/main/java/github/grace000/erversay/RouteHandlers/Put.java:25]: At this point this method has been duplicated for just about every route, right? Have you put any thought into how you could DRY it up?

File: src/main/java/github/grace000/erversay/RouteHandlers/Put.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        for (AcceptedMethods accepted: AcceptedMethods.values()) {
+            if (accepted.name().equals(method)){
+                return true;
+            }
```
Comment (BWoodfork) [inline src/main/java/github/grace000/erversay/RouteHandlers/Put.java:29]: `accepted.name().equals(method)` returns a boolean. You don't need to do a hard return of `true` here.

File: src/test/java/github/grace000/erversay/HandlerTest.java [hunk 3]
```diff
@@ -212,4 +227,61 @@ public void kittyImageHandlerBuildsResponseForImageRequest() {
...
+
+        Request request = new Request("PUT", "/update_body", headers, "update");
+        Response response = putHandler.handle(request);
+
+        String expectedBody = "update";
+        int expectedContent = 6;
+        String expectedHeaders = "Content-Length: 6";
+
+        assertEquals(OK_STATUS, response.status);
+        assert(Objects.deepEquals(expectedBody.getBytes(), response.body));
```
Comment (BWoodfork) [inline src/test/java/github/grace000/erversay/HandlerTest.java:244]: 🥇

File: src/main/java/github/grace000/erversay/RouteHandlers/Put.java (hunk 1 above)
Comment (grace000) [reply src/main/java/github/grace000/erversay/RouteHandlers/Put.java:25]: I could pull that method out into a public helper and use the method across multiple classes.

File: src/main/java/github/grace000/erversay/RouteHandlers/Put.java (hunk 2 above)
Comment (grace000) [reply src/main/java/github/grace000/erversay/RouteHandlers/Put.java:29]: I tried refactoring this before by deleting the hard true, but I received compilation errors.

Comment (BWoodfork) [reply src/main/java/github/grace000/erversay/RouteHandlers/Put.java:29]: Did you do a hard return on the `accepted.name().equals(method)`?

**Pull request merged**